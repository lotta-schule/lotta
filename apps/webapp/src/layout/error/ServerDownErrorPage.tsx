import * as React from 'react';

import { FullErrorPage } from './FullErrorPage';

export interface ServerDownErrorProps {
  error: Error;
}

export const ServerDownErrorPage = React.memo(
  ({ error }: ServerDownErrorProps) => {
    return (
      <FullErrorPage
        title={'Server nicht erreichbar'}
        imageUrl={'/ServerDownImage.svg'}
      >
        <p>
          Der Server hat einen unbekannten Fehler geworfen. Das Team wurde
          informiert.
        </p>
        <p>
          Es tut uns leid f√ºr die Unannehmlichkeiten, hoffentlich funktioniert
          alles in ein paar Minuten wieder.
        </p>
        <p>
          <strong>
            {error.name && <span>{error.name}: </span>}
            <em>{error.message}</em>
          </strong>
        </p>
      </FullErrorPage>
    );
  }
);
ServerDownErrorPage.displayName = 'ServerDownErrorPage';
