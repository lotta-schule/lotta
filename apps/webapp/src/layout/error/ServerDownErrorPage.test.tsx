import { render } from '@testing-library/react';
import { ServerDownErrorPage } from './ServerDownErrorPage';

describe('ServerDownErrorPage', () => {
  it('should render the ServerDownErrorPage component', () => {
    const screen = render(<ServerDownErrorPage error={new Error('Upsi')} />);
    expect(screen.container).toMatchSnapshot();
  });
});
