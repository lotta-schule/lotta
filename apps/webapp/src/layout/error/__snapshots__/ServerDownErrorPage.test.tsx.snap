// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ServerDownErrorPage > should render the ServerDownErrorPage component 1`] = `
<div>
  <section
    class="_error_42c9b3"
  >
    <h1>
      Server nicht erreichbar
    </h1>
    <div>
      <img
        alt="Diese Seite existiert nicht"
        role="presentation"
        src="/ServerDownImage.svg"
      />
      <p>
        Der Server hat einen unbekannten Fehler geworfen. Das Team wurde informiert.
      </p>
      <p>
        Es tut uns leid f√ºr die Unannehmlichkeiten, hoffentlich funktioniert alles in ein paar Minuten wieder.
      </p>
      <p>
        <strong>
          <span>
            Error
            : 
          </span>
          <em>
            Upsi
          </em>
        </strong>
      </p>
    </div>
  </section>
</div>
`;
